forastero/bench.py

```diff
@@ -205,7 +205,7 @@ def register(
         scoreboard: bool = True,
         scoreboard_verbose: bool = False,
         scoreboard_queues: list[str] | None = None,
-    ) -> None:
+    ) -> Component | Coroutine:
```

Comment:
If you want the type system to give the correct return type here you could use overloads:
https://docs.python.org/3/library/typing.html#typing.overload

Could also do with generics but I think the syntax is a bit clunky til 3.12 which makes it nice
https://docs.python.org/3/reference/compound_stmts.html#type-parameter-lists

Reply:
I'll take a read!